### PR TITLE
Factory Design Pattern 

### PR DESCRIPTION
**Pull request by hawashra**

### CODE REVIEW

File: Design Patterns/Abstract Factory Pattern/.idea/.gitignore
Comment (wassiExalt) [inline Design Patterns/Abstract Factory Pattern/.idea/.gitignore:1]: please remove .idea file, apply it for all

File: Design Patterns/Abstract Factory Pattern/Abstract Factory Pattern.iml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (wassiExalt) [inline Design Patterns/Abstract Factory Pattern/Abstract Factory Pattern.iml:11]: what is this file?

File: Design Patterns/Abstract Factory Pattern/src/GPUs/AmdMsiGPU.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package GPUs;
```
Comment (wassiExalt) [inline Design Patterns/Abstract Factory Pattern/src/GPUs/AmdMsiGPU.java:1]: please follow naming convention for packaging: like: com.exalt.training.designpattern

please apply it for all

File: Design Patterns/Abstract Factory Pattern/src/GPUs/AmdMsiGPU.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+public class AmdMsiGPU implements GpuBrand {
```
Comment (wassiExalt) [inline Design Patterns/Abstract Factory Pattern/src/GPUs/AmdMsiGPU.java:3]: there is no need to create a subclass without new fields, different logic, different implementation. so please add a fields, function..etc. to inner class.

File: Design Patterns/Abstract Factory Pattern/src/GPUs/GpuBrand.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package GPUs;
```
Comment (wassiExalt) [inline Design Patterns/Abstract Factory Pattern/src/GPUs/GpuBrand.java:1]: please follow naming convention for packaging: like: com.exalt.training.designpattern

please apply it for all

File: Design Patterns/Abstract Factory Pattern/src/GPUs/GpuBrand.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+
+public interface GpuBrand extends GpuModel {
```
Comment (wassiExalt) [inline Design Patterns/Abstract Factory Pattern/src/GPUs/GpuBrand.java:3]: please add documentation here, and mention the needs for interface.

Comment (wassiExalt) [reply Design Patterns/Abstract Factory Pattern/src/GPUs/GpuBrand.java:3]: apply it for all

File: Design Patterns/Abstract Factory Pattern/src/factories/AsusManufacturer.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
+package factories;
```
Comment (wassiExalt) [inline Design Patterns/Abstract Factory Pattern/src/factories/AsusManufacturer.java:1]: please follow naming convention for packaging: like: com.exalt.training.designpattern

please apply it for all

**Review by wassiExalt — CHANGES_REQUESTED** [review]

File: Design Patterns/Abstract Factory Pattern/src/GPUs/GpuBrand.java (hunk 5 above)
Comment (hawashra) [reply Design Patterns/Abstract Factory Pattern/src/GPUs/GpuBrand.java]: should be done by now.

File: Design Patterns/Abstract Factory Pattern/src/GPUs/AmdMsiGPU.java (hunk 3 above)
Comment (hawashra) [reply Design Patterns/Abstract Factory Pattern/src/GPUs/AmdMsiGPU.java]: the different logic/implementation might be only in the assemble() method implementation but this example is showing a print statement for simplicity. 

I resolved this by mistake but it's not fully resolved yet, I will try to resolve it asap.

File: Design Patterns/Abstract Factory Pattern/.idea/.gitignore
Comment (hawashra) [reply Design Patterns/Abstract Factory Pattern/.idea/.gitignore:1]: should be done by now, I added it to .gitignore and also used git rm -r *.idea and */*.idea to remove it from all directories.

**Review by wassiExalt — APPROVED** [review]

**Pull request merged**